images/warp10/s6/cont-init.d/40-generate-secrets

```diff
@@ -10,10 +10,10 @@ WARP10_CONFIG_DIR="$WARP10_DATA_DIR/conf"
 INITIAL_TOKENS="$WARP10_CONFIG_DIR/initial.token"
 
 if [ ! -f "$INITIAL_TOKENS" ]; then
-    CONFIG_FILES="$(find ${WARP10_CONFIG_DIR} -not -path "*/\.*" -name "*.conf" | sort | tr '\n' ' ' 2> /dev/null)"
+    CONFIG_FILES="$(find ${d} -not -path "*/\.*" -name "*.conf" | sort | tr '\n' ' ' 2> /dev/null)"
 
     # Look for a set token secret and use it for generation
-    secret=`${JAVA} -cp ${WARP10_CP} io.warp10.WarpConfig ${CONFIG_FILES} 'token.secret' | grep 'token.secret' | sed -e 's/^.*=//'`
+    secret=`su ${WARP10_USER} -c "${JAVACMD} -cp ${WARP10_CP} io.warp10.WarpConfig ${CONFIG_FILES} . 'token.secret' | grep -e '^@CONF@ ' | sed -e 's/^@CONF@ //' | grep 'token.secret' | sed -e 's/^.*=//'"`
```

Comment:
Each sed / grep couple can be combined in a single sed with `-n` option (no printing of lines by default) and `s///p` to print sed-ed lines, like:

```
sed -n 's/^@CONF@ //p' | sed -n 's/^token.secret[^=]*=//p'
```

Not sure the syntax is right and not sure about the second command because it depends on the format of the lines, but you get the idea.